src/sort.rs

```diff
@@ -175,19 +184,29 @@ fn sort_by_group(table: &mut Table) {
             .filter(|l| !l.starts_with('#'))
             .count();
 
-        let k = Key::new(&*k).with_leaf_decor(k.leaf_decor().clone());
```

Comment:
This is where the bug was: The whitespace of the key was retained instead of tracked separately for the group index.